src/persist-client/src/internal/trace.rs

```diff
@@ -989,6 +989,8 @@ impl<T: Timestamp + Lattice> Spine<T> {
     /// into a layer which already contains two batches (and is still in the
     /// process of merging).
     fn insert_at(&mut self, batch: Option<SpineBatch<T>>, index: usize) {
+        let Some(batch) = batch else { return };
```

Comment:
I stress a little about this early return diverging. Another approach would be to handle the two cases down in the `self.merging` paths below, which I slightly prefer. Thoughts?

Something like
```
            MergeState::Vacant => match batch {
                Some(batch) => self.merging[index] = MergeState::Single(batch),
                None => self.merging[index] = MergeState::Vacant,
            }
            MergeState::Single(old) => match batch {
                None => self.merging[index] = MergeState::Single(old),
                Some(batch) => {
                    let compaction_frontier = Some(self.since.borrow());
                    self.merging[index] = MergeState::begin_merge(old, batch, compaction_frontier);
                }
            }
```